contract-ui/tabs/overview/cards/custom-contract-code.tsx

```diff
@@ -261,6 +260,66 @@ const InteractiveAbiFunction: React.FC<InteractiveAbiFunctionProps> = ({
   const chainName = getChainName(chainId);
   const [codeEnv, setCodeEnv] = useState<Environment>("javascript");
 
+  async function contractCall(params: unknown[]) {
+    if (!abiFunction) {
+      return undefined;
+    }
+
+    const parsedParams = params.map((p) => {
+      try {
+        const parsed = JSON.parse(p as string);
+        if (Array.isArray(parsed) || typeof parsed === "object") {
+          return parsed;
+        } else {
+          // Return original value if its not an array or object
+          return p;
+        }
+      } catch {
+        // JSON.parse on string will throw an error
+        return p;
+      }
+    })
+    return await contract?.call(abiFunction.name as string, ...parsedParams);
+  }
+
+  function displayString(str: string) {
+    // Only add quotes around string if they aren't already there
+    return str[0] === '"' && str.slice(-1) === '"' ? `${str}` : `"${str}"`;
+  }
+
+  function parseParameter(param: any, language: Environment): string {
+    if (!param.value) {
+      return param.key;
+    }
+
+    // Different syntax for go maps and arrays
+    if (language === "go") {
+      try {
+        const parsed = JSON.parse(param.value);
+        if (Array.isArray(parsed)) {
+          return "[]interface{}{" + parsed.map((p) => parseParameter({ value: JSON.stringify(p) }, "go")).join(", ") + "}";
+        } else if (typeof parsed === "object") {
+          return "map[string]interface{}{" + Object.keys(parsed).map((k) => `"${k}": ${parseParameter({ value: `${parsed[k]}` }, "go")}`).join(", ") +  "}"
+        } {
+          return displayString(param.value);
+        }
+      } catch {
+        return displayString(param.value);
+      }
+    }
+
+    try {
+      const parsed = JSON.parse(param.value);
+      if (Array.isArray(parsed) || typeof parsed === "object") {
+        return param.value;
+      } else {
+        return displayString(param.value);
+      }
+    } catch {
+      return displayString(param.value);
+    }
+  }
```

Comment:
since this functions work directly on their inputs and don't have side-effects can we move them out of the component?

these are basically util functions, they don't need to exist inside the component itself